tests/backup/backup_helper.go

```diff
@@ -3445,9 +3447,56 @@ func DeleteAppNamespace(namespace string) error {
 		return "", false, nil
 	}
 	_, err = task.DoRetryWithTimeout(namespaceDeleteCheck, namespaceDeleteTimeout, jobDeleteRetryTime)
+	if err != nil {
+		ns, err = k8sCore.GetNamespace(namespace)
+		if err != nil {
+			return err
+		}
+		if ns.Status.Phase == "Terminating" {
+			log.Infof("Namespace - %s is in %s phase ", namespace, ns.Status.Phase)
+			err = DeleteTerminatingNamespace(namespace)
+			return err
+		}
+	}
+	return nil
+}
+
+// DeleteTerminatingNamespace deletes the given namespace which has been in terminating state
+func DeleteTerminatingNamespace(namespace string) error {
+	k8sCore := core.Instance()
+	ns, err := k8sCore.GetNamespace(namespace)
 	if err != nil {
 		return err
 	}
+	if ns.Status.Phase == "Terminating" {
+		log.Infof("Namespace - %s is in %s phase ", namespace, ns.Status.Phase)
+		ns.Spec.Finalizers = nil
```

Comment:
How about we log `ns.Spec.Finalizers` before making it nil?

Reply:
Good point. Will log it.